sdk/search/azure-search-documents/azure/search/documents/_service/_generated/models/_models.py

```diff
@@ -3528,311 +3109,1015 @@ def __init__(
+class SearchIndexerDataSource(msrest.serialization.Model):
+    """Represents a datasource definition, which can be used to configure an indexer.
 
     All required parameters must be populated in order to send to Azure.
 
-    :param odata_type: Required. Identifies the concrete type of the token filter.Constant filled
-     by server.
-    :type odata_type: str
-    :param name: Required. The name of the token filter. It must only contain letters, digits,
-     spaces, dashes or underscores, can only start and end with alphanumeric characters, and is
-     limited to 128 characters.
+    :param name: Required. The name of the datasource.
     :type name: str
-    :param pattern: Required. A regular expression pattern.
-    :type pattern: str
-    :param replacement: Required. The replacement text.
-    :type replacement: str
+    :param description: The description of the datasource.
+    :type description: str
+    :param type: Required. The type of the datasource. Possible values include: "azuresql",
+     "cosmosdb", "azureblob", "azuretable", "mysql".
+    :type type: str or ~azure.search.documents.models.SearchIndexerDataSourceType
+    :param credentials: Required. Credentials for the datasource.
+    :type credentials: ~azure.search.documents.models.DataSourceCredentials
+    :param container: Required. The data container for the datasource.
+    :type container: ~azure.search.documents.models.SearchIndexerDataContainer
+    :param data_change_detection_policy: The data change detection policy for the datasource.
+    :type data_change_detection_policy: ~azure.search.documents.models.DataChangeDetectionPolicy
+    :param data_deletion_detection_policy: The data deletion detection policy for the datasource.
+    :type data_deletion_detection_policy:
+     ~azure.search.documents.models.DataDeletionDetectionPolicy
+    :param e_tag: The ETag of the data source.
+    :type e_tag: str
     """
 
     _validation = {
-        'odata_type': {'required': True},
         'name': {'required': True},
-        'pattern': {'required': True},
-        'replacement': {'required': True},
+        'type': {'required': True},
+        'credentials': {'required': True},
+        'container': {'required': True},
     }
 
     _attribute_map = {
-        'odata_type': {'key': '@odata\\.type', 'type': 'str'},
         'name': {'key': 'name', 'type': 'str'},
-        'pattern': {'key': 'pattern', 'type': 'str'},
-        'replacement': {'key': 'replacement', 'type': 'str'},
+        'description': {'key': 'description', 'type': 'str'},
+        'type': {'key': 'type', 'type': 'str'},
+        'credentials': {'key': 'credentials', 'type': 'DataSourceCredentials'},
```

Comment:
The only propery on this - ConnectionString - should actually be on the `SearchIndexerDataSource` type. This isn't passed in by the user, is it?

Reply:
They currently do. This will be a change to the current samples too.